harvester/__main__.py

```diff
@@ -0,0 +1,122 @@
+"""Harvester for netCDF files"""
+
+from datetime import datetime
+from pathlib import Path
+from typing import Union
+
+import click
+import pandas as pd
+
+
+@click.command()
+@click.option(
+    "-d", "--directory", default="./", help="Provide the directory to scan for files"
+)
+@click.option("-o", "--output", default="./", help="Specify the output CSV location")
+@click.option(
+    "-f",
+    "--filename",
+    default="harvested_files.csv",
+    help="Specify the output CSV filename",
+)
+def harvester(
+    directory: Union[Path, str], output: Union[Path, str], filename: Union[Path, str]
+):
+    """
+    Performs the harvesting of netCDF files within a user-defined directory.
+
+    Args:
+        directory (Union[Path, str]): Input directory containing the netCDF files
+        output (Union[Path, str]): Directory
+    """
+    check_directories(directory, output)
+
+    filename_data = scan_directory(directory)
+
+    output_csv(filename_data, output, filename)
```

Comment:
Given that what `output_csv` is doing is pretty simple, I would not bother to add a dedicated function for it. Just do the following and one thing less to test given that we trust that Pandas knows what it is doing:

```suggestion
    filename_data.to_csv(Path(output) / filename, index=False)
```